website/addons/s3/model.py

```diff
@@ -56,6 +57,10 @@ def to_json(self, user):
     def has_auth(self):
         return bool(self.access_key and self.secret_key)
 
+    @property
+    def is_valid(self):
+        return api.has_access(self.access_key, self.secret_key)
```

Comment:
I don't think caching makes sense here. The goal of adding this property is to detect expired and invalid tokens before a user tries to get a list of their S3 buckets. 

Alternatively we can wait until they try to fetch buckets, and then propagate the error from the S3 UI.